galaxy_api/api/ui/serializers/collection.py

```diff
@@ -50,42 +61,46 @@ class CollectionVersionBaseSerializer(serializers.Serializer):
     namespace = serializers.CharField()
     name = serializers.CharField()
     version = serializers.CharField()
-    created_at = serializers.DateTimeField(source='_created')
 
+    created_at = serializers.DateTimeField(source='_created')
 
 class CollectionLatestVersionSerializer(CollectionVersionBaseSerializer):
     metadata = CollectionMetadataBaseSerializer(source='*')
 
+class CollectionLatestVersionDetailSerializer(CollectionLatestVersionSerializer):
+    docs_blob = serializers.JSONField()
+    contents = serializers.ListField(ContentSerializer())
 
 class CollectionVersionSerializer(CollectionMetadataBaseSerializer):
     metadata = CollectionMetadataSerializer(source="*")
-    docs_blob = serializers.JSONField()
 
 
 class _CollectionSerializer(serializers.Serializer):
     id = serializers.UUIDField()
     namespace = serializers.SerializerMethodField()
     name = serializers.CharField()
     download_count = serializers.IntegerField(default=0)
-
     latest_version = CollectionLatestVersionSerializer(source='*')
-    content_summary = ContentSummarySerializer(source='contents')
 
     def _get_namespace(self, obj):
         raise NotImplementedError
 
     def get_namespace(self, obj):
         namespace = self._get_namespace(obj)
-        return NamespaceSerializer(namespace).data
-
+        return NamespaceSummarySerializer(namespace).data
 
 class CollectionListSerializer(_CollectionSerializer):
     def _get_namespace(self, obj):
         name = obj['namespace']
-        return self.context['namespaces'][name]
+        return self.context['namespaces'].get(name, None)
 
 
 class CollectionDetailSerializer(_CollectionSerializer):
+    latest_version = CollectionLatestVersionDetailSerializer(source='*')
+    all_versions = serializers.SerializerMethodField()
```

Comment:
`all_versions` doesn't contain a list of all the versions when the `?version=` param is set.

Reply:
Right, `?version` is a param for filtering the list view which doesn't have `all_versions`